plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2546,6 +2552,62 @@ private static IcebergColumnHandle createProjectedColumnHandle(IcebergColumnHand
                 Optional.empty());
     }
 
+    @Override
+    public Optional<PartialSortApplicationResult<ConnectorTableHandle>> applyPartialSort(
```

Comment:
why not applyTopN?

Reply:
@findepi  File sorting is indeed optional, but necessary for this PR to accelerate queries (we can support partial file sorting scenarios later). Since `table.newScan()` is not recommended, is there another way to get information about whether the files have been sorted during the planning stage ?

And For query acceleration scenarios based on large amounts of top-N data, I think the planning time is negligible ?

Reply:
> why not applyTopN?

To be precise, this PR does not implement push-down top-N, which should be implemented by database software based on JDBC client connections.

This PR rewrites the TopN nodes as follows. I may not have chosen the function name well

```
From:
Top
  Project
    Filter
      TableScan
To:
  Limit
  Project
    Filter
      TableScan
```

Reply:
You are right. In addition, the multi-version of `SortOrder` is not considered now. If the SortOrderId of each file is inconsistent with the SortOrderId defined by the table, then this optimization cannot be used.